main.py

```diff
@@ -1,11 +1,126 @@
 import argparse
+import logging
+import time
+from io import BytesIO
+from PIL import Image
 import logger
 import sound as snd
+import pygame.image
+import camera
 import tcp
-
+import udp
+import struct
 from threading import Thread
 
 buffer_size = None
+INFORMATION_SIZE = struct.calcsize("i" * 3)
+CHUNK_SIZE = 1500 - INFORMATION_SIZE
+
+
+def get_and_send_data(sock, resolution=(640, 480)):
+    sock.settimeout(100)
+    cam = camera.camera_init()
+
+    current_image_number = 0
+    try:
+        while True:
+            while True:
+                if cam.query_image():
+                    camera_image = cam.get_image()
+                    break
+
+            buffer = BytesIO()
+            im = Image.frombuffer("RGB", resolution, bytes(pygame.image.tostring(camera_image, "RGB")))
+            # NOTE: quality parameter was chosen experimentally
+            im.save(buffer, optimize=True, quality=45, format='JPEG')
+            image = buffer.getvalue()
+            buffer.close()
```

Comment:
Is that really necessary? It probably does more harm than good. Garbage collector will free this object anyway, so why bother to spend time freeing it in time-constraint scenario like we have?